indico/modules/events/registration/controllers/display.py

```diff
@@ -392,3 +395,20 @@ def _check_access(self):
     def _process(self):
         filename = secure_filename(f'{self.event.title}-Ticket.pdf', 'ticket.pdf')
         return send_file(filename, generate_ticket(self.registration), 'application/pdf')
+
+
+class RHRegistrationAvatar(RH):
+    """Display a standard avatar for a registration based on the full name."""
+
+    def _process_args(self):
+        self.registration = (Registration
+                             .find(Registration.id == request.view_args['registration_id'],
+                                   ~Registration.is_deleted,
+                                   ~RegistrationForm.is_deleted).one())
```

Comment:
This causes an extremely expensive query since without JOINs it will do a cross-join between the registration form and registration table, so suddenly the DB needs to look at ten thousands of rows (all registration forms that are not deleted multiplied by the number of registrations matching your filter (luckily that's always 1 here)).

Also, `find()` is deprecated ;)

Reply:
```python
class RHRegistrationAvatar(RHDisplayEventBase):
    """Display a standard avatar for a registration based on the full name."""

    normalize_url_spec = {
        'locators': {
            lambda self: self.registration
        }
    }

    def _process_args(self):
        RHDisplayEventBase._process_args(self)
        self.registration = (Registration.query
                             .filter(Registration.id == request.view_args['registration_id'],
                                     ~Registration.is_deleted,
                                     ~RegistrationForm.is_deleted)
                             .join(Registration.registration_form)
                             .one())
```

Here's a similar version that avoids this problem and also enables the url normalization in order to ensure it's not returning a registration from the wrong event.